examples/arkode/F2003_serial/ark_bruss1D_FEM_klu_f2003.f90

```diff
@@ -0,0 +1,1431 @@
+! ------------------------------------------------------------------
+! Programmer(s): David J. Gardner, Cody J. Balos @ LLNL
+!                modified by Jean M. Sexton @ LBL
+!                modified by Daniel M. Margolis @ SMU
+! ------------------------------------------------------------------
+! SUNDIALS Copyright Start
+! Copyright (c) 2002-2023, Lawrence Livermore National Security
+! and Southern Methodist University.
+! All rights reserved.
+!
+! See the top-level LICENSE and NOTICE files for details.
+!
+! SPDX-License-Identifier: BSD-3-Clause
+! SUNDIALS Copyright End
+! ------------------------------------------------------------------
+! Example problem:
+!
+! The following test simulates a brusselator problem from chemical
+! kinetics.  This is a PDE system with 3 components, Y = [u,v,w],
+! satisfying the equations,
+!    du/dt = du*u_xx + a - (w+1)*u + v*u^2
+!    dv/dt = dv*v_xx + w*u - v*u^2
+!    dw/dt = dw*w_xx + (b-w)/ep - w*u
+! for t in the interval [0, 10], x in [0, 10], with initial
+! conditions
+!    u(0,x) =  a  + 0.1*sin(pi*x),
+!    v(0,x) = b/a + 0.1*sin(pi*x),
+!    w(0,x) =  b  + 0.1*sin(pi*x),
+! and with stationary boundary conditions, i.e.
+!    u_t(t,0) = u_t(t,1) = 0,
+!    v_t(t,0) = v_t(t,1) = 0,
+!    w_t(t,0) = w_t(t,1) = 0.
+!
+! Here, we use a piecewise linear Galerkin finite element
+! discretization in space, where all element-wise integrals are
+! computed using 3-node Gaussian quadrature (since we will have
+! quartic polynomials in the reaction terms for the u_t and v_t
+! equations (including the test function)).  The time derivative
+! terms in this system will include a mass matrix, giving rise to
+! an ODE system of the form
+!      M y_t = L y + R(y),
+! where M is the 3x3 block mass matrix for each component, L is
+! the 3x3 block Laplace operator for each component, and R(y) is
+! comprised of the nonlinear reaction terms for each component.
+! Since it it highly inefficient to rewrite this system as
+!      y_t = M^{-1}(L y + R(y)),
+! we solve this system using FARKODE, with a user-supplied mass
+! matrix.  We therefore provide functions to evaluate the ODE RHS
+!    f(t,y) = L y + R(y),
+! its Jacobian
+!    J(t,y) = L + dR/dy,
+! and the mass matrix, M.
+!
+! We use N=201 spatial nodes, with parameters
+!    a=0.6,  b=2.0,  du=0.025,  dv=0.025,  dw=0.025,  ep=1.d-5
+!
+! This program solves the problem with the DIRK method, using a
+! Newton iteration with the SUNKLU sparse linear solvers for both
+! the system and mass matrices.  These matrices are stored in
+! compressed-sparse-row format.
+!
+! Output is printed 10 times throughout the defined time interval.
+! Run statistics (optional outputs) are printed at the end.
+! ------------------------------------------------------------------
+
+module UserData
```

Comment:
Should this module name be more "unique" (to avoid compilation issues with future/other examples that use overly-generic Fortran module names)?